api_fhir/converters/locationConverter.py

```diff
@@ -0,0 +1,153 @@
+        cls.build_fhir_location_address(fhir_location, imis_hf)
+        cls.build_fhir_location_telcome(fhir_location, imis_hf)
+        return fhir_location
+
+    @classmethod
+    def to_imis_obj(cls, fhir_location, audit_user_id):
+        errors = []
+        imis_hf = cls.createDefaultInsuree(audit_user_id)
+        cls.build_imis_hf_identiftier(imis_hf, fhir_location, errors)
+        cls.build_imis_hf_name(imis_hf, fhir_location, errors)
+        cls.build_imis_hf_level(imis_hf, fhir_location, errors)
+        cls.build_imis_hf_address(imis_hf, fhir_location)
+        cls.build_imis_hf_contacts(imis_hf, fhir_location)
+        cls.check_errors(errors)
+        return imis_hf
+
+    @classmethod
+    def createDefaultInsuree(cls, audit_user_id):
+        imis_hf = HealthFacility()
+        # TODO legalForm isn't covered because that value is missing in the model (value need to be nullable in DB)
+        # TODO LocationId isn't covered because that value is missing in the model (value need to be nullable in DB)
+        imis_hf.offline = ApiFhirConfiguration.get_default_value_of_location_offline_attribute()
+        imis_hf.care_type = ApiFhirConfiguration.get_default_value_of_location_care_type()
+        imis_hf.validity_from = core.datetime.datetime.now()
+        imis_hf.audit_user_id = audit_user_id
+        return imis_hf
+
+    @classmethod
+    def build_fhir_location_identifier(cls, fhir_location, imis_hf):
+        identifiers = []
+        cls.build_fhir_id_identifier(identifiers, imis_hf)
+        cls.build_fhir_hf_code_identifier(identifiers, imis_hf)
+        fhir_location.identifier = identifiers
+
+    @classmethod
+    def build_fhir_hf_code_identifier(cls, identifiers, imis_hf):
+        if imis_hf is not None:
+            identifier = cls.build_fhir_identifier(imis_hf.code,
+                                                   ApiFhirConfiguration.get_fhir_identifier_type_system(),
+                                                   ApiFhirConfiguration.get_fhir_facility_id_type())
+            identifiers.append(identifier.__dict__)
+
+    @classmethod
+    def build_imis_hf_identiftier(cls, imis_hf, fhir_location, errors):
+        if fhir_location.get('identifier') is not None:
+            for identifier in fhir_location.get('identifier'):
+                identifier_type = identifier.get("type")
+                if identifier_type is not None \
+                        and identifier_type.get('coding') is not None \
+                        and identifier_type.get('coding')[0].get("system") == ApiFhirConfiguration \
+                        .get_fhir_identifier_type_system():
+                    if identifier_type.get('coding')[0].get("code") == ApiFhirConfiguration.get_fhir_facility_id_type():
+                        if identifier.get("value") is not None:
+                            imis_hf.code = identifier.get("value")
+        cls.valid_condition(imis_hf.code is None, 'Missing hf code', errors)
```

Comment:
I think error codes should be translatable. Is it option to use it as resource string ?

Reply:
Very good point. However I supposed this is out of the scope (changes in existing code is also required). I've created ticket for that (https://openimis.atlassian.net/browse/ONCO-63) and it will be one of next steps.